src/wire_msg.rs

```diff
@@ -0,0 +1,144 @@
+// Copyright 2021MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under the MIT license <LICENSE-MIT
+// https://opensource.org/licenses/MIT> or the Modified BSD license <LICENSE-BSD
+// https://opensource.org/licenses/BSD-3-Clause>, at your option. This file may not be copied,
+// modified, or distributed except according to those terms. Please review the Licences for the
+// specific language governing permissions and limitations relating to use of the SAFE Network
+// Software.
+
+use super::{Error, MsgEnvelope, Result};
+use cookie_factory::{bytes::be_u16, combinator::slice, gen};
+use std::fmt::Debug;
+
+/// Current version of the messaging protocol
+const MESSAGING_PROTO_VERSION: u16 = 1u16;
+
+#[derive(Debug)]
+pub struct WireMsg {
+    hdr: WireMsgHdr,
+    payload: Vec<u8>,
+}
+
+impl WireMsg {
+    pub fn new(msg: &MsgEnvelope, serialisation: PayloadSerType) -> Result<WireMsg> {
+        let payload = match serialisation {
+            PayloadSerType::Json => serde_json::to_string(msg).map_err(|err| {
+                Error::Serialisation(format!(
+                    "Could not serialize message payload (id: {}): {}",
+                    msg.id(),
+                    err
+                ))
+            })?,
+            PayloadSerType::Bincode => "bincode".to_string(), // TODO
+        };
+
+        let hdr = WireMsgHdr {
+            version: MESSAGING_PROTO_VERSION,
+            payload_ser: serialisation,
+        };
+
+        Ok(Self {
+            hdr,
+            payload: payload.as_bytes().to_vec(),
+        })
+    }
+
+    pub fn from(buf: &[u8]) -> Result<Self> {
+        // Let's read the servialisation protocol version bytes first
+        let mut version_bytes = [0; 2];
+        version_bytes[0..].copy_from_slice(&buf[0..2]);
+        // TODO: return error if version is not supported
+        let version = u16::from_be_bytes(version_bytes);
+
+        // ...now read the type of serialisation used for the payload
+        let mut ser_type_bytes = [0; 2];
```

Comment:
```suggestion
        let mut serialisation_type_bytes = [0; 2];
```
and elsewhere here too. I was confused at first read of this with just `ser`

Reply:
thanks, i also was trying with diff things but was hesitating with these naming, i'll update it.